client/post-editor/editor-ground-control/index.jsx

```diff
@@ -294,29 +292,32 @@ export class EditorGroundControl extends PureComponent {
 						/>
 						{ this.getVerificationNoticeLabel() }{' '}
 						<span className="editor-ground-control__email-verification-notice-more">
-							{ this.props.translate( 'Learn More' ) }
+							{ translate( 'Learn More' ) }
 						</span>
 					</div>
 				) }
-				<div className="editor-ground-control__status">
-					{ this.isSaveEnabled() && (
-						<button
-							className="editor-ground-control__save button is-link"
-							onClick={ this.onSaveButtonClick }
-							tabIndex={ 3 }
-						>
-							{ this.props.translate( 'Save' ) }
-						</button>
-					) }
-					{ ! this.isSaveEnabled() && (
-						<span
-							className="editor-ground-control__save-status"
-							data-e2e-status={ this.getSaveStatusLabel( identity ) }
-						>
-							{ this.getSaveStatusLabel( this.props.translate ) }
-						</span>
-					) }
-				</div>
+				{ ( isSaveEnabled || shouldShowStatusLabel ) && (
+					<div className="editor-ground-control__status">
```

Comment:
I'd quite like to pull this out in to a separate component file, happy to do so if it seems valuable.

Reply:
I could go either way on making it a separate component. If there are other parts of the codebase that are already doing something similar and it could be reused by them, that that'd probably make a good case. Otherwise, 🤷‍♂️ .

If I were writing it, I would probably at least pull this out into a separate function though, similar to `renderGroundControlActionButtons()`. The logic that's currently in there to control what gets displays feels a bit too cumbersome for something inside `render()`, at least for my tastes.

It feels like it might also be able to have a simpler flow inside a normal JS function, maybe with something like this (not tested):

```js
renderStatusEleemnts() {
		if ( ! isSaveEnabled && ! shouldShowStatusLabel ) {
			return;
		}

		let elements = [];

		if ( isSaveEnabled ) {
			elements.push = (
				<button
					className="editor-ground-control__save button is-link"
					onClick={ this.onSaveButtonClick }
					tabIndex={ 3 }
				>
					{ translate( 'Save' ) }
				</button>
			);
		}

		if ( ! isSaveEnabled && shouldShowStatusLabel ) {
			elements.push = (
				<span
						className="editor-ground-control__save-status"
						data-e2e-status={ isSaving ? 'Saving…' : 'Saved' }
					>
						{ isSaving ? translate( 'Saving…' ) : translate( 'Saved' ) }
					</span>
			);
		}

		return (
			<div className="editor-ground-control__status">
				{ elements }
			</div>
		);
	}
```

There are probably some ways to clean that up, but it's kinda the direction I'd head.

That's a nitpick, though, and probably personal preference to some degree. I wouldn't be surprised if other devs preferred the current approach you took.

Reply:
> I would probably at least pull this out into a separate function though

I tend to err towards the idea that if it can be pulled in to a function, it can be pulled in to a new component.

> The logic that's currently in there to control what gets displays feels a bit too cumbersome for something inside render

Agreed! I like to see a chunk of JSX that doesn't take too much visual parsing to figure out what'll happen.

I'm going to try breaking this section out.